src/test/rules/no-classfield-shadowing_test.ts

```diff
@@ -0,0 +1,167 @@
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    sourceType: 'module',
+    ecmaVersion: 'latest'
+  }
+});
+
+const parser = require.resolve('@babel/eslint-parser');
+const parserOptions = {
+  requireConfigFile: false,
+  babelOptions: {
+    plugins: [
+      ['@babel/plugin-proposal-decorators', {decoratorsBeforeExport: true}]
+    ]
+  }
+};
+
+ruleTester.run('no-classfield-shadowing', rule, {
+  valid: [
+    `class MyElement extends LitElement {
+      static properties = {
+        foo: { type: String }
+      }
+    }`,
+    `class MyElement extends LitElement {
+      foo;
+      properties = {
+        foo: { type: String }
+      }
+    }`
+  ],
+
+  invalid: [
+    {
+      code: `class MyElement extends LitElement {
+        foo;
+        static properties = {foo: {}}
+      }`,
+      errors: [
+        {
+          messageId: 'noClassfieldShadowing',
+          data: {prop: 'foo'},
+          line: 3,
+          column: 30
+        }
+      ]
+    },
+    {
+      code: `class MyElement extends LitElement {
+        static properties = {foo: {}}
+        foo;
+      }`,
+      errors: [
+        {
+          messageId: 'noClassfieldShadowing',
+          data: {prop: 'foo'},
+          line: 2,
+          column: 30
+        }
+      ]
+    },
+    {
+      code: `class MyElement extends LitElement {
+        foo;
+        static get properties() { return { foo: {}}};
+      }`,
+      errors: [
+        {
+          messageId: 'noClassfieldShadowing',
+          data: {prop: 'foo'},
+          line: 3,
+          column: 44
+        }
+      ]
+    },
+    {
+      code: `class MyElement extends LitElement {
+        static get properties() { return { foo: {}}};
+        foo;
+      }`,
+      errors: [
+        {
+          messageId: 'noClassfieldShadowing',
+          data: {prop: 'foo'},
+          line: 2,
+          column: 44
+        }
+      ]
+    },
+    {
+      code: `class Foo extends A(LitElement) {
+        foo;
+        static properties = { foo: {} };
+      }`,
+      errors: [
+        {
+          messageId: 'noClassfieldShadowing',
+          data: {prop: 'foo'},
+          line: 3,
+          column: 31
+        }
+      ]
+    },
+    {
+      code: `class Foo extends A(B(LitElement)) {
+        foo;
+        static properties = { foo: {} };
+      }`,
+      errors: [
+        {
+          messageId: 'noClassfieldShadowing',
+          data: {prop: 'foo'},
+          line: 3,
+          column: 31
+        }
+      ]
+    },
+    {
+      code: `class Foo extends A(B(C(LitElement))) {
+        foo;
+        static properties = { foo: {} };
+      }`,
+      errors: [
+        {
+          messageId: 'noClassfieldShadowing',
+          data: {prop: 'foo'},
+          line: 3,
+          column: 31
+        }
+      ]
+    },
+    {
+      code: `class Foo extends LitElement {
+        @property({ type: String })
+        foo = 'test';
+
+        static properties = { foo: {} };
```

Comment:
this seems like an odd case. the lit docs aren't so clear, are we meant to use `declare` for decorated properties?

if i understood correctly, if you use decorators you only have two options:

- use `declare`
- use `useDefineForClassFields: false`

so this test is actually behaving the same as if it didn't have a decorator, right? and we need another once we have typescript parser support in tests (in another pr)

Reply:
i would remove it and we can re-add it once we have the typescript parser in here 👍